src/components/DownloadLimitEntry.vue

```diff
@@ -0,0 +1,186 @@
+  - GNU Affero General Public License for more details.
+  -
+  - You should have received a copy of the GNU Affero General Public License
+  - along with this program. If not, see <http://www.gnu.org/licenses/>.
+  -
+-->
+
+<template>
+	<Fragment>
+		<NcCheckboxRadioSwitch :checked.sync="limitEnabled"
+			:loading="loading"
+			:disabled="hasError">
+			{{ t('files_downloadlimit', 'Limit downloads') }}
+		</NcCheckboxRadioSwitch>
+		<template v-if="!loading && !hasError">
+			<NcNoteCard v-show="limitEnabled && showRemainingDownloadsNote"
+				class="action__count-note"
+				type="info">
+				{{ n('files_downloadlimit', '1 remaining download allowed', '{count} remaining downloads allowed', remainingCount, { count: remainingCount }) }}
+			</NcNoteCard>
+			<NcTextField v-show="limitEnabled"
+				:label="t('files_downloadlimit', 'Set download limit')"
+				type="number"
+				min="1"
+				:value.sync="limit"
+				:helper-text="helperText"
+				:error="Boolean(helperText)" />
+			<NcNoteCard v-show="limitEnabled && showResetNote"
+				class="action__reset-note"
+				type="warning">
+				{{ t('files_downloadlimit', 'Setting a new limit will reset the download count') }}
+			</NcNoteCard>
+		</template>
+	</Fragment>
+</template>
+
+<script lang="ts">
+import { defineComponent } from 'vue'
+import { Fragment } from 'vue-frag'
+import { showError } from '@nextcloud/dialogs'
+
+import NcCheckboxRadioSwitch from '@nextcloud/vue/dist/Components/NcCheckboxRadioSwitch.js'
+import NcNoteCard from '@nextcloud/vue/dist/Components/NcNoteCard.js'
+import NcTextField from '@nextcloud/vue/dist/Components/NcTextField.js'
+
+import {
+	getDownloadLimit,
+	setDownloadLimit,
+	deleteDownloadLimit,
+} from '../services/DownloadLimitService.ts'
+import { logger } from '../logger.ts'
+
+export default defineComponent({
+	name: 'DownloadLimitEntry',
+
+	components: {
+		Fragment,
+		NcCheckboxRadioSwitch,
+		NcNoteCard,
+		NcTextField,
+	},
+
+	props: {
+		share: {
+			type: Object,
+			required: true,
+		},
+	},
+
+	data() {
+		return {
+			limitEnabled: false,
+			initialLimit: null,
+			limit: '', // input should be empty at first so initialize as empty string
+			count: null,
+			loading: false,
+			hasError: false,
+		}
+	},
+
+	computed: {
+		remainingCount() {
+			return this.initialLimit - this.count
+		},
+
+		helperText() {
+			if (this.limit > 0) {
+				return ''
+			}
+			return t('files_downloadlimit', 'The minimum limit is 1')
+		},
+
+		isNewShare() {
+			return this.share.id === null || this.share.id === undefined
+		},
+
+		showRemainingDownloadsNote() {
+			return typeof this.count === 'number'
+		},
+
+		showResetNote() {
+			return !this.isNewShare
+				&& Boolean(this.initialLimit)
+				&& typeof this.limit === 'number'
+				&& this.limit !== this.initialLimit
+		},
+	},
+
+	watch: {
+		limit(limit) {
+			this.limit = Number(limit) // emitted <input> value is string so we parse it to number
+		},
+	},
```

Comment:
I'd suggest using a dedicated listener instead of `value.sync` to cast the string to an int. Something like `@update:value="onLimitUpdate"` ? :thinking: 
The watcher seems a bit too hacky to my taste.

Reply:
https://github.com/nextcloud/files_downloadlimit/pull/230